valhalla/baldr/double_bucket_queue.h

```diff
@@ -68,6 +93,11 @@ class DoubleBucketQueue final {
     // Allocate the low-level buckets
     size_t bucketcount = (range / bucketsize_) + 1;
     buckets_.resize(bucketcount);
+    for (auto& b : buckets_) {
+      b.clear();
+      b.reserve(bucketsize_ / 2 + 1);
```

Comment:
What is the value of reserving here? The bucketsize is generally = 1 (meaning each bucket covers all edges that have cost within 1 second of each other). Maybe bucketsize is a misleading name but it is intended to mean the cost range held within a bucket. Ideally you would estimate the number of edges that might end up within the same bucket - but that is highly dependent on the density of the network touched by the actual route so I can't think of a metric to intelligently reserve an appropriate size. Many buckets could actually end up being empty so reserving a larger vector may be counter-productive.

Reply:
Thank you, @dnesbitt61. I misunderstood meaning of bucketsize_. Perhaps you can suggest some better value for reserve, then just not reserving anything?

Reply:
Well, the problem is that some buckets may be more populated than others (majority are likely to be 0). I think one would have to do some routes and log the sizes that occur (max size, perhaps the % of buckets that have any edges added). If this is only ever used in server applications then reserving a modest size for each bucket (32, 64?) makes sense but what about embedded applications? I'd hate to reserve a bunch of memory for sparsely populated buckets. Perhaps this behavior can be made optional so embedded applications can better control memory use?

Reply:
According to experiment with test set of routes on desktop, this reserve wins statistically insignificant performance improvement. Average value is 0.0, maximum value encountered was 14, but usually it's no more then 4. Requested bucketcount during the experiment was 2501 or 20001.

Reply:
`b.reserve(...)` is removed in last update of this PR